### DISCUSSION TITLE
How to add a Forgejo

### DISCUSSION THREAD

**Discussion by papanito** [ImranR98/Obtainium#1989] [Q&A] [answered]
What are rhr exact steps to add a Forgejo repo. I have for example https://git.citron-emu.org/Citron/Citron/releases/download/nightly-20241118_094118/Citron-Android.apk as a basis but adding this url or part of it always ends in "Not found"

**Comment by DarkCrypt** [comment, marked as answer]
Adding this app is basically the same as adding an app from GitHub. I'll provide a config, including a breakdown, so you have an idea how to do so for future apps.

**Edit:**
Citron has 2 APKs with 2 different app ID's.

Citron (Canary Refresh) config (app-mainline-release_optimised):
Import App -> [Citron (Canary Refresh)](https://apps.obtainium.imranr.dev/redirect?r=obtainium://app/%7B%22id%22%3A%22com.miHoYo.Yuanshen%22%2C%22url%22%3A%22https%3A%2F%2Fgit.citron-emu.org%2FCitron%2FCitron%22%2C%22author%22%3A%22Citron%22%2C%22name%22%3A%22citron%22%2C%22preferredApkIndex%22%3A0%2C%22additionalSettings%22%3A%22%7B%5C%22includePrereleases%5C%22%3Atrue%2C%5C%22fallbackToOlderReleases%5C%22%3Atrue%2C%5C%22filterReleaseTitlesByRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22filterReleaseNotesByRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22verifyLatestTag%5C%22%3Afalse%2C%5C%22dontSortReleasesList%5C%22%3Afalse%2C%5C%22useLatestAssetDateAsReleaseDate%5C%22%3Afalse%2C%5C%22releaseTitleAsVersion%5C%22%3Afalse%2C%5C%22trackOnly%5C%22%3Afalse%2C%5C%22versionExtractionRegEx%5C%22%3A%5C%22v%5B0-9%5D%2B%5C%5C%5C%5C.%5B0-9%5D%2B%5C%22%2C%5C%22matchGroupToUse%5C%22%3A%5C%22%5C%22%2C%5C%22versionDetection%5C%22%3Afalse%2C%5C%22useVersionCodeAsOSVersion%5C%22%3Afalse%2C%5C%22apkFilterRegEx%5C%22%3A%5C%22optimised%5C%22%2C%5C%22invertAPKFilter%5C%22%3Afalse%2C%5C%22autoApkFilterByArch%5C%22%3Atrue%2C%5C%22appName%5C%22%3A%5C%22Citron%20(Canary%20Refresh)%5C%22%2C%5C%22shizukuPretendToBeGooglePlay%5C%22%3Afalse%2C%5C%22allowInsecure%5C%22%3Afalse%2C%5C%22exemptFromBackgroundUpdates%5C%22%3Afalse%2C%5C%22skipUpdateNotifications%5C%22%3Afalse%2C%5C%22about%5C%22%3A%5C%22A%20cutting-edge%20Nintendo%20Homebrew%20emulator%20designed%20to%20deliver%20an%20optimized%20experience%20for%20playing%20your%20favorite%20games%20and%20exploring%20new%20ones.%5C%22%2C%5C%22refreshBeforeDownload%5C%22%3Afalse%7D%22%2C%22overrideSource%22%3A%22Codeberg%22%7D)
- URL: `https://git.citron-emu.org/Citron/Citron` (base URL)
- Override Source: `Forgejo (Codeberg)` (Obtainium didn't automatically choose this source)
- Include prereleases: `Toggle ON` (the releases are tagged pre-release)
- Trim Version String With ReGex: `v[0-9]+\.[0-9]+` (trims unnecessary text to only get version number)
- Filter APKs by Regular Expression: `optimised` (targets this APK)
- App Name: `Citron (Canary Refresh)` (added name, otherwise it just says citron)
- About: `A cutting-edge Nintendo Homebrew emulator designed to deliver an optimized experience for playing your favorite games and exploring new ones.`
- Leave all other settings as default

Citron (Canary Refresh) config (app-mainline-release):
Import App -> [Citron (Canary Refresh)](https://apps.obtainium.imranr.dev/redirect?r=obtainium://app/%7B%22id%22%3A%22org.citron.citron_emu%22%2C%22url%22%3A%22https%3A%2F%2Fgit.citron-emu.org%2FCitron%2FCitron%22%2C%22author%22%3A%22Citron%22%2C%22name%22%3A%22citron%22%2C%22preferredApkIndex%22%3A0%2C%22additionalSettings%22%3A%22%7B%5C%22includePrereleases%5C%22%3Atrue%2C%5C%22fallbackToOlderReleases%5C%22%3Atrue%2C%5C%22filterReleaseTitlesByRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22filterReleaseNotesByRegEx%5C%22%3A%5C%22%5C%22%2C%5C%22verifyLatestTag%5C%22%3Afalse%2C%5C%22dontSortReleasesList%5C%22%3Afalse%2C%5C%22useLatestAssetDateAsReleaseDate%5C%22%3Afalse%2C%5C%22releaseTitleAsVersion%5C%22%3Afalse%2C%5C%22trackOnly%5C%22%3Afalse%2C%5C%22versionExtractionRegEx%5C%22%3A%5C%22v%5B0-9%5D%2B%5C%5C%5C%5C.%5B0-9%5D%2B%5C%22%2C%5C%22matchGroupToUse%5C%22%3A%5C%22%5C%22%2C%5C%22versionDetection%5C%22%3Afalse%2C%5C%22useVersionCodeAsOSVersion%5C%22%3Afalse%2C%5C%22apkFilterRegEx%5C%22%3A%5C%22release%5C%5C%5C%5C.%5C%22%2C%5C%22invertAPKFilter%5C%22%3Afalse%2C%5C%22autoApkFilterByArch%5C%22%3Atrue%2C%5C%22appName%5C%22%3A%5C%22Citron%20(Canary%20Refresh)%5C%22%2C%5C%22shizukuPretendToBeGooglePlay%5C%22%3Afalse%2C%5C%22allowInsecure%5C%22%3Afalse%2C%5C%22exemptFromBackgroundUpdates%5C%22%3Afalse%2C%5C%22skipUpdateNotifications%5C%22%3Afalse%2C%5C%22about%5C%22%3A%5C%22A%20cutting-edge%20Nintendo%20Homebrew%20emulator%20designed%20to%20deliver%20an%20optimized%20experience%20for%20playing%20your%20favorite%20games%20and%20exploring%20new%20ones.%5C%22%2C%5C%22refreshBeforeDownload%5C%22%3Afalse%7D%22%2C%22overrideSource%22%3A%22Codeberg%22%7D)
- Filter APKs by Regular Expression: `release\.`
*Same as above except for the APK filter.

**Reply by papanito** [reply]
many thanks

**Comment by dougsund** [comment]
I just re-added citron to obtainium (android)  with this commented link and I'm getting the unexpected error "RangeError (start): Invalid Value: not in inclusive range 0..1: -1

Also I can't install citron from obtainium, the install button is greyed out. if I do a manual install, then the update function doesn't work, I get the same error.  

Can anyone assist with this one?
